manifest-build-tools/application/reprove.py

```diff
@@ -195,6 +214,167 @@ def directory_for_repo(self, repo):
         repo_directory = os.path.join(self._builddir, repo_directory)
         return repo_directory
 
+    def branch_existing_repositories(self):
+        """
+        Issues create branch commands to repos in a provided manifest
+        :return: None
+        """
+        if self._branch_name is None:
+            print "Please provide the new branch name"
+            sys.exit(2)
+
+        repo_list = self._manifest.get_repositories()
+        if repo_list is None:
+            print "No repository list found in manifest file"
+            sys.exit(2)
+        else:
+            # Loop through list of repos and create specified branch on each
+            for repo in repo_list:
+                self.create_repo_branch(repo, self._branch_name)
+
+    def create_repo_branch(self, repo, branch):
+        """
+        create branch  on the repos in the manifest file
+        :param repo: A dictionary
+        :return: None
+        """
+        try:
+            repo_directory = self.directory_for_repo(repo)
+            repo_url = repo["repository"]
+       
+            self.repo_operator.create_repo_branch(repo_url, repo_directory, branch)
+
+        except RuntimeError as error:
+            print "Exiting due to error: {0}".format(error)
+            sys.exit(1)   
+    
+    def checkout_branch_repositories(self, branch):
+        repo_list = self._manifest.get_repositories()
+        if repo_list is None:
+            print "No repository list found in manifest file"
+            sys.exit(2)
+        else:
+            # Loop through list of repos and checkout specified branch on each
+            for repo in repo_list:
+                self.checkout_repo_branch(repo, branch)
+
+    def checkout_repo_branch(self, repo, branch):
```

Comment:
There is different between the function checkout() in repo_operator  and this function.
The function checkout() in repo_operator creates a new branch under a directory.
But this function creates branch according to a dictionary.

I don't think it's better to combine this function to checkout_branch_repositories.
Because two methods simplify the function of a method.
And we have no idea whether we would need to checkout branch for one repository.